lib/ddtrace/sampler.rb

```diff
@@ -175,8 +199,8 @@ def priority_assigned_upstream?(span)
       span.context && !span.context.sampling_priority.nil?
     end
 
-    def priority_sample(span)
-      @priority_sampler.sample?(span)
+    def priority_sample!(span)
+      @priority_sampler.sample!(span)
```

Comment:
We change this to use `sample!` instead of `sample?` to allow for `@priority_sampler` to set additional tags that are unrelated to priority sampling.
In our case, we want to set '_dd.rule_psr' and  '_dd.limit_psr' when sampling happens.

Reply:
Gotcha, makes sense.